cdc/sink/mq/dispatcher/partition/index_value.go

```diff
@@ -14,13 +14,16 @@
 package partition
 
 import (
+	"sync"
+
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/pkg/hash"
 )
 
 // IndexValueDispatcher is a partition dispatcher which dispatches events based on the index value.
 type IndexValueDispatcher struct {
 	hasher *hash.PositionInertia
+	lock   sync.Mutex
```

Comment:
Could you do a benchmark here proving that lock contention will not be a bottleneck?

Reply:
Sure. I'll test it.

Reply:
I think the impact in this area is limited, in this PR I am moving the locks from the original BufferSink to the dispatcher level. Maybe a real cluster test is needed to determine the impact on it.

Reply:
See: https://github.com/pingcap/tiflow/pull/5302#issuecomment-1126655847